container-disc/src/main/java/com/yahoo/container/jdisc/component/Deconstructor.java

```diff
@@ -61,9 +81,24 @@ public void deconstruct(List<Object> components, Collection<Bundle> bundles) {
                 ((SharedResource) component).release();
             }
         }
-        if (! destructibleComponents.isEmpty() || ! bundles.isEmpty())
-            executor.schedule(new DestructComponentTask(destructibleComponents, bundles),
+        if (!destructibleComponents.isEmpty() || !bundles.isEmpty()) {
+            var task = executor.schedule(new DestructComponentTask(destructibleComponents, bundles),
                               delay.getSeconds(), TimeUnit.SECONDS);
+            if (mode.equals(Mode.SHUTDOWN)) {
+                // Wait for deconstruction to finish
+                try {
+                    log.info("Waiting up to " + SHUTDOWN_DECONSTRUCT_TIMEOUT.toSeconds() + " seconds for all components to deconstruct.");
```

Comment:
Isn't there also a 50s timeout on deconstruct, somewhere further up?

Reply:
Yes, you're right. That seems a bit harsh. Perhaps we should increase it to at least 90 sec, and set this one to ~80?

Reply:
I think 45s is good!